src/ast.rs

```diff
@@ -339,6 +339,7 @@ pub struct FunctionDecl {
     /// the boolean indicates whether to bind by-reference (true)
     pub usev: Vec<(bool, RcStr)>,
     pub ret_ref: bool,
+    pub ret_type: Option<RcStr>
```

Comment:
Ah I didn't consider that when writing the comment.
Use `NullableTy` insteadof `Ty` (it already exists)